src/python/WMCore/REST/Validation.py

```diff
@@ -26,22 +30,22 @@ def _check_rx(argname, val, custom_err = None):
 
 def _check_str(argname, val, rx, custom_err = None):
     """
-    convert unicode to str first to support cherrypy 3.2.2
+    convert unicode to str (bytes) first to support cherrypy 3.2.2
     This is not really check val is ASCII.
     """
-    if isinstance(val, unicode):
+    if isinstance(val, str):
         try:
-            val = str(val)
+            val = bytes(val, "utf-8")
         except:
-            raise InvalidParameter(return_message("Invalid '%s' parameter" % argname, custom_err))
+            raise InvalidParameter(return_message("Invalid '%s' parameter %s %s" % (argname, val, type(val)), custom_err))
     if not isinstance(val, basestring) or not rx.match(val):
         raise InvalidParameter(return_message("Incorrect '%s' parameter" % argname, custom_err))
     return val
 
 def _check_ustr(argname, val, rx, custom_err = None):
-    if isinstance(val, str):
+    if isinstance(val, bytes):
```

Comment:
I'm sort of concerned with these back and forth with bytes/strings. If nothing was caught in the unit tests, that's a first good sign though.

Reply:
`__check_ustr()` is used by `validate_ustr()`, which in the docstring states

> Validates that an argument is a string and matches a regexp, once converted from utf-8 into unicode.

So I just convert a bytes string to unicode. This is one of the few cases in which the intent of the code is clear to me and I am pretty confident that this will not break anything. Let's hope I am right!